joomla/joomla-cms
# [4.1] mod_breadcrumbs - convert microdata to JSON-LD

Pull Request for Issue # .

### Summary of Changes

With https://github.com/joomla/joomla-cms/pull/25117 in mind and the possible adaptation of Spatie\SchemaOrg in mind I would like to prepare current breadcrumbs by converting the Microdata into JSON-LD structure. 
But not only with that possible adaptation in mind... The JSON-LD format is the recommended format. 
The HTML code will be cleaner when the data is moved to JSON-LD format
Another benefit of this change is that the current page will always be added to the data. This is not the case in the current situation with _Hide Last Item_ is active. 


### Testing Instructions

- Install Joomla with Sample Data Blog. 
- Note that there will be a  mod_breadcrumbs on every page. example: http://**your-joomla-4-site**/index.php/blog/article
- Inspect element and see the markup of mod_breadcrumbs
- Apply this patch. No `npm run something:something` is required
- Refresh the page and note the changed markup.


### Actual result BEFORE applying this Pull Request

```
<nav role="navigation" aria-label="Breadcrumbs">
    <ol itemscope="" itemtype="https://schema.org/BreadcrumbList" class="mod-breadcrumbs breadcrumb">
        <li class="mod-breadcrumbs__here float-left">You are here: &nbsp;</li>
        <li itemprop="itemListElement" itemscope="" itemtype="https://schema.org/ListItem" class="mod-breadcrumbs__item breadcrumb-item">
            <a itemprop="item" href="/index.php" class="pathway"><span itemprop="name">Home</span></a>
            <meta itemprop="position" content="1">
        </li>
        <li itemprop="itemListElement" itemscope="" itemtype="https://schema.org/ListItem" class="mod-breadcrumbs__item breadcrumb-item">
            <a itemprop="item" href="/index.php/blog" class="pathway"><span itemprop="name">Blog</span></a>
            <meta itemprop="position" content="2">
        </li>
        <li aria-current="page" itemprop="itemListElement" itemscope="" itemtype="https://schema.org/ListItem" class="mod-breadcrumbs__item breadcrumb-item active">
            <span itemprop="name">Article</span>
            <meta itemprop="position" content="3">
        </li>
    </ol>
</nav>

```


### Expected result AFTER applying this Pull Request

```
<nav role="navigation" aria-label="Breadcrumbs">
    <ol class="mod-breadcrumbs breadcrumb">
        <li class="mod-breadcrumbs__here float-left">You are here: &nbsp;</li>
        <li class="mod-breadcrumbs__item breadcrumb-item"><a href="/index.php" class="pathway"><span>Home</span></a></li>
        <li class="mod-breadcrumbs__item breadcrumb-item"><a href="/index.php/blog" class="pathway"><span>Blog</span></a></li>
        <li class="mod-breadcrumbs__item breadcrumb-item active"><span>Article</span></li>
    </ol>
</nav>
```

While inspecting the element and scroll to top and inspect element `<head>`. It should contain the following `<script>` tag. 
```
    <script type="application/ld+json">
        {
            "@context": "https:\/\/schema.org",
            "@type": "BreadcrumbList",
            "itemListElement": [
                {
                    "@type": "ListItem",
                    "position": 1,
                    "item": {
                        "@id": "http:\/\/joomla4.test\/index.php",
                        "name": "Home"
                    }
                },
                {
                    "@type": "ListItem",
                    "position": 2,
                    "item": {
                        "@id": "http:\/\/joomla4.test\/index.php\/blog",
                        "name": "Blog"
                    }
                },
                {
                    "@type": "ListItem",
                    "position": 3,
                    "item": {
                        "@id": "http:\/\/joomla4.test\/index.php\/blog\/article",
                        "name": "Article"
                    }
                }
            ]
        }    </script>
```


### Documentation Changes Required
